tests/components/lupusec/test_config_flow.py

```diff
@@ -0,0 +1,233 @@
+        data=MOCK_DATA_STEP,
+        unique_id=MOCK_DATA_STEP[CONF_HOST],
+    )
+
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        MOCK_DATA_STEP,
+    )
+
+    await hass.async_block_till_done()
+
+    assert result2["type"] == FlowResultType.ABORT
+    assert result2["reason"] == "already_configured"
+
+
+@pytest.mark.parametrize(
+    ("mock_import_step", "mock_title"),
+    [
+        (MOCK_IMPORT_STEP, MOCK_IMPORT_STEP[CONF_IP_ADDRESS]),
+        (MOCK_IMPORT_STEP_NAME, MOCK_IMPORT_STEP_NAME[CONF_NAME]),
+    ],
+)
+async def test_flow_source_import(
+    hass: HomeAssistant, mock_import_step, mock_title
+) -> None:
+    """Test configuration import from YAML."""
+    with patch(
+        "homeassistant.components.lupusec.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry, patch(
+        "homeassistant.components.lupusec.config_flow.lupupy.Lupusec.__init__",
+        return_value=None,
+    ) as mock_initialize_lupusec:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data=mock_import_step,
+        )
+
+    await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == mock_title
+    assert result["data"] == MOCK_DATA_STEP
+    assert len(mock_setup_entry.mock_calls) == 1
+    assert len(mock_initialize_lupusec.mock_calls) == 1
+
+
+@pytest.mark.parametrize(
+    ("raise_error", "text_error"),
+    [
+        (LupusecException("Test lupusec exception"), "cannot_connect"),
+        (Exception("Test unknown exception"), "unknown"),
+    ],
+)
+async def test_flow_source_import_error_and_recover(
+    hass: HomeAssistant, raise_error, text_error
+) -> None:
+    """Test exceptions and recovery."""
+
+    with patch(
+        "homeassistant.components.lupusec.config_flow.lupupy.Lupusec.__init__",
+        side_effect=raise_error,
+    ) as mock_initialize_lupusec:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data=MOCK_IMPORT_STEP,
+        )
+
+    await hass.async_block_till_done()
+    assert result["type"] == FlowResultType.ABORT
+    assert result["reason"] == text_error
+    assert len(mock_initialize_lupusec.mock_calls) == 1
+
+
+async def test_flow_source_import_already_configured(hass: HomeAssistant) -> None:
+    """Test duplicate config entry.."""
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        title=MOCK_DATA_STEP[CONF_HOST],
+        data=MOCK_DATA_STEP,
+        unique_id=MOCK_DATA_STEP[CONF_HOST],
```

Comment:
Our config entry doesn't have this as unique id anymore

Reply:
Good find :)